src/modules/campath/mod.rs

```diff
@@ -0,0 +1,350 @@
+//! Campath
+
+use std::path::PathBuf;
+
+use self::bvh::{read_bvh, Bvh};
+use self::camio::CamIO;
+use self::exporter::Exporter;
+use super::commands::Command;
+use super::Module;
+use crate::handler;
+use crate::hooks::engine::{self, con_print};
+use crate::modules::campath::camio::read_camio;
+use crate::modules::cvars::CVar;
+use crate::utils::*;
+
+mod bvh;
+mod camio;
+mod common;
+mod exporter;
+
+pub struct Campath;
+impl Module for Campath {
+    fn name(&self) -> &'static str {
+        "Campath"
+    }
+
+    fn description(&self) -> &'static str {
+        "Loading and exporting campath motion .cam or .bvh by HLAE and its Blender addon."
+    }
+
+    fn commands(&self) -> &'static [&'static Command] {
+        static COMMANDS: &[&Command] = &[
+            &BXT_CAMPATH_FORCE_LIVE,
+            &BXT_CAMPATH_EXPORT_START,
+            &BXT_CAMPATH_EXPORT_STOP,
+        ];
+        COMMANDS
+    }
+
+    fn cvars(&self) -> &'static [&'static CVar] {
+        static CVARS: &[&CVar] = &[&BXT_CAMPATH_OFFSET, &BXT_CAMPATH_LOAD];
+        CVARS
+    }
+
+    fn is_enabled(&self, marker: MainThreadMarker) -> bool {
+        engine::CL_PlayDemo_f.is_set(marker)
+            && engine::CL_Disconnect.is_set(marker)
+            && engine::CL_ViewDemo_f.is_set(marker)
+            && engine::cls_demos.is_set(marker)
+            && engine::host_frametime.is_set(marker)
+            && engine::r_refdef_vieworg.is_set(marker)
+            && engine::r_refdef_viewangles.is_set(marker)
+            && engine::R_RenderView.is_set(marker)
+            && engine::scr_fov_value.is_set(marker)
+            && engine::V_RenderView.is_set(marker)
+    }
+}
+
+static BXT_CAMPATH_LOAD: CVar = CVar::new(
+    b"bxt_campath_load\0",
+    b"\0",
+    "\
+Loads campath .cam or .bvh file and playback during demo.
+",
+);
+
+static BXT_CAMPATH_OFFSET: CVar = CVar::new(
+    b"bxt_campath_offset\0",
+    b"-0.25\0",
+    "\
+Offsets time values in second of each campath entry. Offset can be negative. Default is -0.25.
```

Comment:
> Default for offset is -0.25 because of loading time.

Why? Is this somehow not recorded during recording but played back during playback?

Reply:
The loading time is when you load the campath as soon as the demo/map starts. When you record a campath, it is not necessary that you are recording as soon as the map starts. So this offset only happens when loading demo. It is more than often that campath is played during demo playback, so this offset is for that.